platform/linuxbsd/detect.py

```diff
@@ -111,15 +112,30 @@ def configure(env):
             env["CXX"] = "clang++"
         env.extra_suffix = ".llvm" + env.extra_suffix
 
-    if env["use_lld"]:
-        if env["use_llvm"]:
-            env.Append(LINKFLAGS=["-fuse-ld=lld"])
-            if env["use_thinlto"]:
-                # A convenience so you don't need to write use_lto too when using SCons
-                env["use_lto"] = True
+    if env["linker"] != "default":
+        print("Using linker program: " + env["linker"])
+        if env["linker"] == "mold" and using_gcc(env):  # GCC < 12.1 doesn't support -fuse-ld=mold.
+            cc_version = get_compiler_version(env)
+            cc_semver = (int(cc_version["major"]), int(cc_version["minor"]))
+            if cc_semver < (12, 1):
+                if os.path.isfile("/usr/libexec/mold/ld"):
+                    env.Append(LINKFLAGS=["-B/usr/libexec/mold"])
```

Comment:
Mold isn't always in `/usr/libexec` or `/usr/local/libexec`. [Arch Linux](https://archlinux.org/packages/community/x86_64/mold/files/) and [KISS Linux](https://github.com/kiss-community/repo/blob/0251a738d447c776054e490677cee91381956261/extra/mold/build#L18) have it in `/usr/lib`.

Reply:
Well Arch has latest GCC so it wont use this codepath :P 

But we won't be able to satisfy all distro whims, this is just a convenience feature for the most expected setup (documented by upstream). Others will just have to use `mold -run scons`.

Still, I guess I can add `/usr/lib` too for the distros which for some reason choose not to use `/usr/libexec` for what it's designed for.